### PR TITLE
Enhance `subdatasets` command to set values in .gitmodules

### PR DESCRIPTION
**Pull request by mih**
While discussing #1549 we came up with the need to label subdatasets as 'no-auto-install' (or something like this). `subdatasets` already reports all values from `.gitmodules`, but we lack the ability to set them conveniently.

This PR adds this feature.

- [x] dissolve `rewrite_urls` when `subdatasets` get's the ability to digest value template strings.

### CODE REVIEW

File: datalad/distribution/subdatasets.py [hunk 1]
```diff
@@ -257,10 +303,36 @@ def _get_submodules(dspath, fulfilled, recursive, recursion_limit,
...
+            submodule_section = 'submodule "{}"'.format(sm['gitmodule_name'])
+            # first deletions
+            for dprop in assure_list(delete_property):
+                parser.remove_option(submodule_section, dprop)
+                # also kick from the info we just read above
+                sm.pop('gitmodule_{}'.format(dprop), None)
+            # and now setting values
+            for sprop in assure_list(set_property):
+                prop, val = sprop
+                if '{' in val:
```
Comment (yarikoptic) [inline datalad/distribution/subdatasets.py]: Isn't { a legit character for a filename/path? I just wonder if we should just rely on some custom prefix to say that it is a template and should go through format

Comment (yarikoptic) [inline datalad/distribution/subdatasets.py]: Not to say that we might want to have some property which would contain { in its value and might not want to format it here

Comment (mih) [inline datalad/distribution/subdatasets.py]: Any suggestions for a prefix? I don't care, I just used the same mechanism that we use for --output-format.

Comment (mih) [inline datalad/distribution/subdatasets.py]: OK, added additional requirement for a template to be wrapped in `<>` and contain `{` before expansion is attempted`.

**Pull request merged**